components/director/internal/domain/runtime/service.go

```diff
@@ -377,42 +369,30 @@ func (s *service) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
-// SetLabel missing godoc
+// SetLabel sets Runtime label from a given input
 func (s *service) SetLabel(ctx context.Context, labelInput *model.LabelInput) error {
 	rtmTenant, err := tenant.LoadFromContext(ctx)
 	if err != nil {
 		return errors.Wrapf(err, "while loading tenant from context")
 	}
 
-	err = s.ensureRuntimeExists(ctx, rtmTenant, labelInput.ObjectID)
-	if err != nil {
+	if err = s.ensureRuntimeExists(ctx, rtmTenant, labelInput.ObjectID); err != nil {
 		return err
 	}
 
-	currentRuntimeLabels, err := s.getCurrentLabelsForRuntime(ctx, rtmTenant, labelInput.ObjectID)
-	if err != nil {
+	if modifiable, err := isLabelModifiable(labelInput.Key, s.protectedLabelPattern, s.immutableLabelPattern); err != nil {
 		return err
+	} else if !modifiable {
+		return apperrors.NewInvalidDataError("could not set unmodifiable label with key %s", labelInput.Key)
 	}
 
-	newRuntimeLabels := make(map[string]interface{})
-	for k, v := range currentRuntimeLabels {
-		newRuntimeLabels[k] = v
-	}
-
-	newRuntimeLabels[labelInput.Key] = labelInput.Value
-
-	if err = s.upsertScenariosLabelIfShould(ctx, labelInput.ObjectID, labelInput.Key, newRuntimeLabels); err != nil {
-		return err
-	}
+	id := labelInput.ObjectID
 
-	modifiable, err := isLabelModifiable(labelInput.Key, s.protectedLabelPattern, s.immutableLabelPattern)
-	if err != nil {
-		return err
-	}
-	if !modifiable {
-		return apperrors.NewInvalidDataError("could not set unmodifiable label with key %s", labelInput.Key)
-	}
-	if labelInput.Key != model.ScenariosKey {
+	if labelInput.Key == model.ScenariosKey {
+		if err := s.updateScenariosLabel(ctx, rtmTenant, id, map[string]interface{}{model.ScenariosKey: labelInput.Value}); err != nil {
```

Comment:
you could just return s.updateScenariosLabel(...). Then 'if err' and the else clause wont be needed